.circleci/config.yml

```diff
@@ -38,34 +77,52 @@ jobs:
             docker compose -f docker/docker-compose.yml \
                            -f docker/docker-compose.dev.yml \
                            -f docker/docker-compose.postgres.yml \
-                           --env-file .env.dev up -d \
-                           --build
+                           --env-file .env.dev up -d
       - run:
           name: Test Postgres
           command: |
             set -x
             source ~/project/docker/commands.bash
             cdash_run_and_submit_pgsql_ctest
-      - run:
-          name: Build Production Docker Images
-          command: |
-              DOCKER_TAG="${CIRCLE_TAG:-latest}"
-              docker build -t kitware/cdash:${DOCKER_TAG} --target cdash .
-              docker build -t kitware/cdash-worker:${DOCKER_TAG} --target cdash-worker .
+
+  Publish Images:
+    environment:
+      SITENAME: CircleCI
+    machine:
+      image: ubuntu-2204:current
+      docker_layer_caching: true
+    steps:
+      - checkout
       - run:
           name: Publish Docker Images
           command: |
-            if [ "${CIRCLE_BRANCH}" == "master" ] || [ -n "${CIRCLE_TAG}" ]; then
-              DOCKER_TAG="${CIRCLE_TAG:-latest}"
+            if [ "${CIRCLE_BRANCH}" == "master" ]; then
               echo "$DOCKER_PASS" | docker login -u "$DOCKER_USER" --password-stdin
               docker push kitware/cdash:testing
-              docker push kitware/cdash:${DOCKER_TAG}
-              docker push kitware/cdash-worker:${DOCKER_TAG}
+              docker push kitware/cdash:latest
+              docker push kitware/cdash-worker:latest
+            fi
+
+            if [ -n "${CIRCLE_TAG}" ]; then
+              echo "$DOCKER_PASS" | docker login -u "$DOCKER_USER" --password-stdin
+              docker push kitware/cdash:${CIRCLE_TAG}
+              docker push kitware/cdash-worker:${CIRCLE_TAG}
             fi
+
 workflows:
   all-builds:
     jobs:
-      - build:
-          filters:
-            tags:
```

Comment:
Question: Where did this functionality go?  I don't think I read the equivalent of "run when a tag is pushed" in the new side.  Or is that we're set up to run more often and it'll be covered?

Otherwise, I don't have any suggestions to make!

Reply:
We were confused about what that did.  After doing some more research, I think that's required to trigger the CI when a tag is pushed and I have added it back.  It's worth noting that with these new changes, the image is built and pushed to Docker Hub, but the CI is not run.  My thought was that there is no point in running the tests when a tag is pushed since we run the CI on every change via the merge queue anyway.  Removing the tests from the pushing process will speed up the release process and make it easier for me to make releases.